src/bp/bpfs.ts

```diff
@@ -0,0 +1,229 @@
+    }
+  }
+
+  async pullChanges() {
+    const cleanBefore = process.argv.includes('--clean')
+    const axiosClient = this._getPullAxiosClient()
+    const dryRun = process.argv.includes('--dry')
+
+    try {
+      // We clear only those two folders, so assets are preserved
+      if (cleanBefore) {
+        console.log(chalk.blue(`Cleaning data folder before pulling data...`))
+        await this._clearDir(path.join(this.targetDir, 'global'))
+        await this._clearDir(path.join(this.targetDir, 'bots'))
+      } else if (fse.existsSync(this.targetDir)) {
+        const fileCount = await this._filesCount(this.targetDir)
+        console.log(chalk.blue(`Remote files will be pulled in an existing folder containing ${fileCount} files`))
+      }
+
+      console.log(chalk.blue(`Pulling all remote changes from ${this.serverUrl} to ${this.targetDir} ...`))
+
+      const { data: archive } = await axiosClient.get('export')
+      if (dryRun) {
+        console.log(chalk.yellow(`Dry run completed. Successfully downloaded archive.`))
+        return process.exit()
+      }
+
+      console.log(
+        chalk.blue(`Extracting archive to local file system... (archive size: ${bytesToString(archive.length)}`)
+      )
+
+      await extractArchive(archive, this.targetDir)
+
+      const fileCount = await this._filesCount(this.targetDir)
+      console.log(chalk.green(`Successfully extracted ${fileCount} files from the remote server!`))
+    } catch (err) {
+      const error = err.response ? `${err.response.statusText} (${err.response.status})` : err.message
+      this._endWithError(`Could not pull changes: ${error}`)
+    }
+  }
+
+  async pushChanges() {
+    const useForce = process.argv.includes('--force')
+    const dryRun = process.argv.includes('--dry')
+
+    if (!fse.existsSync(this.sourceDir)) {
+      this._endWithError(`Specified folder "${this.sourceDir}" doesn't exist.`)
+    }
+
+    try {
+      console.log(chalk.blue(`Preparing an archive of your local files from ${this.sourceDir}...`))
+
+      const archive = await createArchiveFromFolder(this.sourceDir, pushedArchiveIgnoredFiles)
+      const axiosClient = this._getPushAxiosClient(archive.length)
+
+      console.log(
+        chalk.blue(`Sending archive to server for comparison... (archive size: ${bytesToString(archive.length)})`)
+      )
+      const { data } = await axiosClient.post('changes', archive)
+      const { changeList, blockingChanges, localFiles } = this._processChanges(data)
+
+      if (_.isEmpty(blockingChanges) || useForce) {
+        this._printChangeList(changeList)
+
+        if (dryRun) {
+          console.log(chalk.yellow(`Dry run completed. Nothing was pushed to server`))
+          return process.exit()
+        }
+
+        console.log(chalk.blue(`Pushing local changes to ${this.serverUrl}... ${useForce ? '(using --force)' : ''}`))
+
+        await axiosClient.post('update', archive)
```

Comment:
oh I see client is used here as well.. maybe my other comment isn't valid

Reply:
I'm using the client twice for different task (one to list changes, and the other to commit them), and I really tried to make them as small as possible, the max there could be is to save 1 or 2 lines